### ISSUE TITLE
Secure Settings Permission Grant Failure in Beta 2

### ISSUE THREAD

**Issue by mattjanicki** [Android-L-Porting-Team/Android-L-Mako#27]
I use the app [Secure Settings](https://play.google.com/store/apps/details?id=com.intangibleobject.securesettings.plugin&hl=en) for performing certain tasks with Tasker (switching to Airplane Mode, locking the device, setting or unsetting a PIN) and it requires root permissions to install something called a "System+ Module" in order to perform certain functions.  Superuser fails to grant the appropriate permissions for this module to be installed. I've included a [pastebin of the relevant logcat](http://pastebin.com/DgSNahxe) in case it helps. I don't know if it's related to Superuser or something specific to the port, but I figured I'd mention it as a bug.

**Comment by Darex2094** [comment]
I'm assuming you're using the bundled Superuser instead of SuperSU or anything like that. Can you pastebin the log from Superuser as well? I'd like to take a look at why it's denying the grant.

**Comment by mattjanicki** [comment]
Yup, I'm using the bundled Superuser.  Do you know which directory the Superuser logs are saved in?  I checked /data/data/com.koushikdutta.superuser and can't seem to find them, nor is there any obvious directory on /sdcard/.

**Comment by Darex2094** [comment]
Well, actually, before digging into that, can you go into Superuser and see if it even saw the request? Should show a grant/deny right there. If we see a deny I'll look at where it keeps logs because off the top of my head I don't know.

**Comment by mattjanicki** [comment]
Yup, seen and granted according to Superuser.  It's Secure Settings that says the permission was denied.
![screenshot_2014-07-08-12-01-30](https://cloud.githubusercontent.com/assets/3886387/3512483/90dffe68-06b9-11e4-8dfd-0c80c77b2b86.png)

**Comment by Darex2094** [comment]
Okay... interesting. May I ask what build you're using?

I'm going to file a report upstream and see if I can find something out. I'll link it here once the bug report is up in case you want to follow along.

EDIT: Actually, scratch that. I send a bug report upstream with a root issue and I'm sure some folks will have a chuckle. Let me find out where Superuser keeps it's logs at (assuming it doesn't just log like that with nothing else)

**Comment by mattjanicki** [comment]
Yeah, I too initially assumed it was a basic superuser issue, but everything else appears to work fine with Superuser and I used the same Superuser build with Secure Settings on 4.4.4 without issue.  I'm thinking it might be a directory tree issue in the port (since the System+ module is looking to install the module as a system app), but that's purely conjecture.

**Comment by Darex2094** [comment]
I can't find anything about logs Superuser stores -- except for the ones screenshotted above... I'm submitting a feature request upstream, though I suspect the reason why is because Superuser just accepts input from the user and also forks based on preset allow/deny lists, so it itself isn't really doing anything worthy of logging beyond what we see above.

This is speculation, and considering I'm not an ANDROID dev I'll need to defer to someone else's experience (I'm a Gentoo Linux dev who's been hiding under a rock when it comes to SELinux, though then again I just got around to accepting that pulseaudio is a good thing and desktop managers aren't the devil anymore, even though they are), but perhaps it's related to SELinux somehow? I'm merely speculating, as this is why I've avoided SELinux like the plague. To me it's a blanket cover-all for stupidity on the internet.

**Comment by mattjanicki** [comment]
I suppose that's possible, but wasn't Kit Kat already in SELinux-enforcing mode?  Shouldn't it be approximately the same?  I Googled around to see whether people on the Nexus 5 developer preview were having the same issue (to determine whether it's Nexus 4 port-specific or an L developer preview issue), but I couldn't find anything related.

**Comment by benpye** [comment]
I don't know how SELinux would be affected this, it's set to permissive, logging but actually has no effect. Has this occurred on the Nexus 5 or Nexus 7 2013?

**Comment by mattjanicki** [comment]
Yeah, that's what I wish I knew.  Obviously if it isn't a port issue, then there's no point in posting this as a bug, but I don't have either the Nexus 5 or Nexus 7 2013 so I have no way of verifying that.

**Comment by Sykopompos** [comment]
4.4.3 was permissive, 4.4.4 was only some things se enforced while on L it is a complete lockdown on almost everything

**Comment by mehdilh423** [comment]
beta 3-pre5 have bug in permissins...deny all app :(
i cant root it

**Comment by maxpayn02** [comment]
I have the same problem , supersu log said that a supersu pro is required then after i installed it the problem remain and the log says : pm grant  com.intangibleobject.securesettings.pluging android.permission.WRITE_secure_settings /system/bin/pm[6] : app_process : PERMISSION DENIED exit

**Comment by leonkim70** [comment]
I had the same problem.
but I had success that.
please, upgrade to 'superSU'(recency version) you will be fix this probleam.

**Comment by Garric15** [comment]
Same for me. I updated superSU like @leonkim70  said to fix the problem.

**Comment by aaronsomek** [comment]
I had this issue with a fresh AndroidN Preview install. I installed busybox and updated the binaries and it fixed the issue.
